deployment/src/test/java/io/quarkiverse/fx/deployment/RunOnFxThreadTest.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkiverse.fx.deployment;
+
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicReference;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.event.Observes;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkiverse.fx.FxApplication;
+import io.quarkiverse.fx.PrimaryStage;
+import io.quarkiverse.fx.RunOnFxThread;
+import io.quarkus.test.QuarkusUnitTest;
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+public class RunOnFxThreadTest {
+
+    private static final int LAUNCH_TIMEOUT_MS = 1_000;
+    public static final String JAVA_FX_APPLICATION_THREAD = "JavaFX Application Thread";
```

Comment:
Simplify with Platform.isFxApplicationThread()
And maybe we have no guaranty that the thread name could change is next versions